Neos.Media.Browser/Classes/Controller/AssetProxyController.php

```diff
@@ -0,0 +1,168 @@
+<?php
+
+namespace Neos\Media\Browser\Controller;
+
+/*
+ * This file is part of the Neos.Media.Browser package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+  *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Media\Domain\Model\AssetSource\AssetNotFoundException;
+use Neos\Media\Domain\Model\AssetSource\AssetProxy\HasRemoteOriginal;
+use Neos\Media\Domain\Model\AssetSource\AssetProxy\SupportsIptcMetadata;
+use Neos\Media\Domain\Model\AssetSource\AssetSource;
+use Neos\Media\Domain\Model\AssetSource\AssetSourceConnectionException;
+use Neos\Media\Domain\Model\ImportedAsset;
+use Neos\Media\Domain\Model\ImportedAssetManager;
+use Neos\Media\Domain\Model\AssetSource\AssetSourceAwareInterface;
+use Neos\Media\Domain\Repository\ImportedAssetRepository;
+use Neos\Flow\Annotations as Flow;
+use Neos\Flow\Mvc\Controller\ActionController;
+use Neos\Flow\ResourceManagement\Exception;
+use Neos\Flow\ResourceManagement\ResourceManager;
+use Neos\Media\Domain\Model\Asset;
+use Neos\Media\Domain\Repository\AssetRepository;
+use Neos\Media\Domain\Strategy\AssetModelMappingStrategyInterface;
+
+/**
+ * @Flow\Scope("singleton")
+ */
+class AssetProxyController extends ActionController
+{
+    /**
+     * @Flow\InjectConfiguration(path="assetSources", package="Neos.Media")
+     * @var array
+     */
+    protected $assetSourcesConfiguration;
+
+    /**
+     * @var AssetSource[]
+     */
+    protected $assetSources = [];
+
+    /**
+     * @Flow\Inject
+     * @var ResourceManager
+     */
+    protected $resourceManager;
+
+    /**
+     * @Flow\Inject
+     * @var AssetRepository
+     */
+    protected $assetRepository;
+
+    /**
+     * @Flow\Inject
+     * @var ImportedAssetRepository
+     */
+    protected $importedAssetRepository;
+
+    /**
+     * @Flow\Inject
+     * @var AssetModelMappingStrategyInterface
+     */
+    protected $assetModelMappingStrategy;
+
+    /**
+     * @Flow\Inject
+     * @var ImportedAssetManager
+     */
+    protected $importedAssetManager;
+
+    /**
+     * @return void
+     */
+    public function initializeObject()
+    {
+        foreach ($this->assetSourcesConfiguration as $assetSourceIdentifier => $assetSourceConfiguration) {
```

Comment:
This stuff could be DRY'd and extracted from here and the other controller.

Reply:
"Three strikes and you refactor" is the rule I usually apply.